lib/active_force/bulk/records.rb

```diff
@@ -16,10 +18,25 @@ def to_csv
       end
 
       def self.parse_from_attributes(records)
+        # Sorting ensures that the headers line up with the values for the CSV
         headers = records.first.keys.sort.map(&:to_s)
-        data = records.map { |r| r.sort.pluck(-1) }
+        data = records.map do |r|
+           r.transform_values { |v| transform_value_for_sf(v) }.sort.pluck(-1)
+        end
         new(headers: headers, data: data)
       end
+
+      # SF expects a special value for setting a column to be NULL.
+      def self.transform_value_for_sf(value)
+        case value
+        when NilClass
+          NULL_VALUE
+        when Time
```

Comment:
Does this need to be DateTime and Time? What about just Date.

Reply:
After further conversation - it looks like DateTime is already handled through calling the method: 

`    def translate_to_sf attributes
      attrs = attributes.map do |attribute, value|
        field = fields[attribute.to_sym]
        [field.sfdc_name, field.value_for_hash(value)]
      end
      Hash[attrs]
    end

 `
 
 LGTM!